src/v/cloud_storage/tests/partition_manifest_test.cc

```diff
@@ -613,23 +643,80 @@ struct segment_meta_v0 {
 
     auto operator<=>(const segment_meta_v0&) const = default;
 };
+struct segment_meta_v1 {
+    using value_t = segment_meta_v1;
+    static constexpr serde::version_t redpanda_serde_version = 2;
+    static constexpr serde::version_t redpanda_serde_compat_version = 0;
```

Comment:
what's going on with this structure? why is it a serde structure versioned separately as v1?